targets/arty/net.py

```diff
@@ -1,53 +1,50 @@
-from litex.soc.integration.soc_core import mem_decoder
 from litex.soc.integration.soc_sdram import *
 
 from liteeth.core.mac import LiteEthMAC
 from liteeth.phy import LiteEthPHY
 
-from targets.utils import csr_map_update
-from targets.arty.base import SoC as BaseSoC
+from targets.utils import dict_set_max
+from .base import BaseSoC
 
 
 class NetSoC(BaseSoC):
-    csr_peripherals = (
-        "ethphy",
-        "ethmac",
-    )
-    csr_map_update(BaseSoC.csr_map, csr_peripherals)
-
-    mem_map = {
-        "ethmac": 0x30000000,  # (shadow @0xb0000000)
-    }
-    mem_map.update(BaseSoC.mem_map)
+    mem_map = {**BaseSoC.mem_map, **{
+        "ethmac": 0xb0000000,
+    }}
 
     def __init__(self, platform, *args, **kwargs):
         # Need a larger integrated ROM on or1k to fit the BIOS with TFTP support.
-        if 'integrated_rom_size' not in kwargs and kwargs.get('cpu_type', 'lm32') != 'lm32':
-            kwargs['integrated_rom_size'] = 0x10000
+        if kwargs.get('cpu_type', 'lm32') != 'lm32':
+            dict_set_max(kwargs, 'integrated_rom_size', 0x10000)
 
         BaseSoC.__init__(self, platform, *args, **kwargs)
 
+        # Ethernet ---------------------------------------------------------------------------------
+        # Ethernet PHY
         self.submodules.ethphy = LiteEthPHY(
             platform.request("eth_clocks"),
             platform.request("eth"))
-        self.submodules.ethmac = LiteEthMAC(
-            phy=self.ethphy, dw=32, interface="wishbone", endianness=self.cpu.endianness)
-        self.add_wb_slave(mem_decoder(self.mem_map["ethmac"]), self.ethmac.bus)
-        self.add_memory_region("ethmac",
-            self.mem_map["ethmac"] | self.shadow_base, 0x2000)
+        self.add_csr("ethphy")
 
-        self.ethphy.crg.cd_eth_rx.clk.attr.add("keep")
-        self.ethphy.crg.cd_eth_tx.clk.attr.add("keep")
-        #self.platform.add_period_constraint(self.crg.cd_sys.clk, 10.0)
-        self.platform.add_period_constraint(self.ethphy.crg.cd_eth_rx.clk, 40.0)
-        self.platform.add_period_constraint(self.ethphy.crg.cd_eth_tx.clk, 40.0)
+        # Ethernet MAC
+        ethmac_win_size = 0x2000
+        self.submodules.ethmac = LiteEthMAC(
+            phy        = self.ethphy,
+            dw         = 32,
+            interface  = "wishbone",
+            endianness = self.cpu.endianness)
+        self.add_wb_slave(self.mem_map["ethmac"], self.ethmac.bus, ethmac_win_size)
+        self.add_memory_region("ethmac", self.mem_map["ethmac"], ethmac_win_size, type="io")
```

Comment:
There was some difference between `add_mem_region` and `add_memory_region` -- Can you check with @enjoy-digital what the right form is?

Reply:
`add_mem_region` has been removed from LiteX in https://github.com/enjoy-digital/litex/commit/8bc420679a3d7dbb4c89ec5e110e4e606d088752 - now only `add_memory_region` version is available.